### PR TITLE
add some logic for known properties

### PR DESCRIPTION
**Pull request by incertia**
makes #1649 go brrr. i haven't updated the examples yet but would something like this be ok?

```
- **Breaking** Known physical device properties (VkPhysicalDeviceProperties, VkPhysicalDeviceLimits, VkPhysicalDeviceSparseProperties) are now no longer `Option<T>`.
  - These properties are available with every `vkGetPhysicalDeviceProperties2` call
- Add some `Default` instances to make the auto trait happy
```

### CODE REVIEW

**Comment by Rua** [comment]
While it's funny, unfortunately GitHub doesn't understand "makes #1649 go brrr" to link the two issues. :D

**Comment by incertia** [comment]
any ideas for improving the macro would be greatly appreciated. i had to parse it as `tt` because `literal` would not work and had to duplicate some code in the macros but as long as it works i guess.

**Comment by Rua** [comment]
The autogen module is essentially already a macro, it generates code. So what you could do is eliminate the `properties!` macro entirely, and generate the final code entirely in autogen. That gives you much more freedom, because you are generating code with full Rust instead of the macro language.

File: vulkano/src/version.rs [hunk 1]
```diff
@@ -39,6 +39,12 @@ impl Version {
     }
 }
 
+impl Default for Version {
```
Comment (Eliah-Lakhin) [inline vulkano/src/version.rs:42]: Please fix the indentation.

**Comment by Eliah-Lakhin** [comment]
@incertia Thank you for your work! This improvement looks good to me in general. Please fix the issue with indentation I mentioned above, and if possible maybe get rid of macro use(however, I can proceed with merging anyway, the improvement is useful!)

**Comment by incertia** [comment]
done

**Comment by incertia** [comment]
> The autogen module is essentially already a macro, it generates code. So what you could do is eliminate the properties! macro entirely, and generate the final code entirely in autogen. That gives you much more freedom, because you are generating code with full Rust instead of the macro language.

I think this should be addressed in a separate PR. It would be quite a large change but if you think we can do it here I can look into making the changes as well.

**Comment by Rua** [comment]
It's fine. If you want to go ahead and do it, you can, but it's no problem if you leave this PR as it is. I or someone else can do the conversion some other time.

**Comment by incertia** [comment]
Probably good to merge then. I'll look into emitting the correct code within autogen sometime this week I hope.

**Comment by Eliah-Lakhin** [comment]
@incertia Pull Request merged. Thanks again! I have a number of pending PRs to review, so the review process is taking time, but I hope to clean up the rest soon too.

**Pull request merged**